core/perf_test/PerfTest_Stream.cpp

```diff
@@ -0,0 +1,284 @@
+
+template <unsigned MemTraits>
+using StreamView = Kokkos::View<double*, Kokkos::MemoryTraits<MemTraits>>;
+
+// different than benchmarks/stream, which uses int
+// wide index types are common as GPU memory grows
+using StreamIndex = int64_t;
+using Policy      = Kokkos::RangePolicy<Kokkos::IndexType<StreamIndex>>;
+
+template <typename V>
+void perform_set(const V& a, const double scalar) {
+  Kokkos::parallel_for(
+      "set", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { a[i] = scalar; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_copy(const V& a, const V& b) {
+  Kokkos::parallel_for(
+      "copy", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { b[i] = a[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_scale(const V& b, const V& c, const double scalar) {
+  Kokkos::parallel_for(
+      "scale", Policy(0, b.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { b[i] = scalar * c[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_add(const V& a, const V& b, const V& c) {
+  Kokkos::parallel_for(
+      "add", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { c[i] = a[i] + b[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_triad(const V& a, const V& b, const V& c, const double scalar) {
+  Kokkos::parallel_for(
+      "triad", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { a[i] = b[i] + scalar * c[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+int validate_array(V& a_dev, const double expected) {
+  const auto a = Kokkos::create_mirror_view(Kokkos::WithoutInitializing, a_dev);
+  Kokkos::deep_copy(a, a_dev);
+
+  double error = 0.0;
+  for (size_t i = 0; i < a.size(); ++i) {
+    error += std::abs(a[i] - expected);
+  }
+  const double avgError = error / (double)a.size();
+
+  constexpr double epsilon = 1.0e-13;
+  return std::abs(avgError / expected) > epsilon;
+}
+
+template <unsigned MemTraits>
+static void StreamSet(benchmark::State& state) {
+  const size_t N8                 = std::pow(state.range(0), 8);
+  static constexpr int DATA_RATIO = 1;
+
+  StreamView<MemTraits> a(Kokkos::view_alloc(Kokkos::WithoutInitializing, "a"),
+                          N8);
+
+  for (auto _ : state) {
+    Kokkos::Timer timer;
+    perform_set(a, SCALAR);
+    KokkosBenchmark::report_results(state, a, DATA_RATIO, timer.seconds());
+  }
+
+  if (validate_array(a, SCALAR)) {
+    state.SkipWithError("validation failure");
+  }
+}
+
+template <unsigned MemTraits>
+static void StreamCopy(benchmark::State& state) {
+  const size_t N8                 = std::pow(state.range(0), 8);
+  static constexpr int DATA_RATIO = 2;
+
+  StreamView<MemTraits> a(Kokkos::view_alloc(Kokkos::WithoutInitializing, "a"),
+                          N8);
+  StreamView<MemTraits> b(Kokkos::view_alloc(Kokkos::WithoutInitializing, "b"),
+                          N8);
```

Comment:
I am not familiar with the STREAM benchmark.  Is it expected to act on raw allocated memory?

Reply:
@maartenarnst do you have a strong feeling either way?

Reply:
Because we're overwriting the data, I assumed that the memory being raw memory wouldn't have an impact. @dalg24, are you suggesting that that assumption is not right and we should expect an impact?